### ISSUE TITLE
force replace resource problem

### ISSUE THREAD

**Issue by juliostone** [newrelic/terraform-provider-newrelic#2046]
Guys, good afternoon! I would like your support in a scenario that we are facing. 
We are currently using the **notification_channels** features to deliver alerts related to our policies at the infrastructure level throughout the environment, however when we access the old notification settings interface, we receive the message that the feature is at end of life and will be discontinued in 25 and 31st of this month and what existing resources will be migrated to the **workflow** solution
 While testing the new solution with the official newrelic provider for terraform (our official IaC tool at the moment) we came across some bugs to modify the **newrelic_workflow** resource that would make it impossible to migrate the infrastructure as current code to the new solution.
The first error scenario is related to changes within the workflow filter block. Any change in the post creation resource filter block causes the error message below:
**Error: maximum retries reached: An error occurred resolving this field**
We are also facing a scenario in which when we change the destination channels linked to the workflow, the notification channel is deleted from the new relic interface, but without being deleted from the terraform inventory file, causing the automations to break, as follows terraform below in case you need to reproduce the problem
fyi: @leonardoHaubrich

```
terraform {
  backend "local" {
    path = "terraform_tmp.tfstate"
  }
  required_providers {
    newrelic = {
      source  = "newrelic/newrelic"
      version = "3.3.0"
    }
  }
}

provider "newrelic" {
  account_id = var.accountId
  api_key    = var.apiKey # usually prefixed with 'NRAK'
  region     = "US"                               # Valid regions are US and EU
}

resource "newrelic_alert_policy" "policy1" {
  name = "nr-workflow-test-1"
}

resource "newrelic_alert_policy" "policy2" {
  name = "nr-workflow-test-2"
}

resource "newrelic_notification_destination" "default-slack" {
  name = "company"
  type = "SLACK"
  property {
    key= "scope"
    label= "Permissions"
    value= "chat:write,incoming-webhook,users:read,users:read.email,app_mentions:read,channels:join,channels:read,chat:write.public,commands,groups:read,links:read,links:write,team:read"
  }
  property {
    key   = "teamName"
    label = "Team Name"
    value = "company"
  }
  auth_token {
    prefix= "Bearer"
    token= ""
  }
}

# pager dutty integration
resource "newrelic_notification_destination" "default-pagerduty" {
  account_id = var.accountId
  name       = "pagerduty-tmp-fin-sre"
  type       = "PAGERDUTY_SERVICE_INTEGRATION"

  property {
    key   = ""
    value = ""
  }

  auth_token {
    prefix = "Token token="
    token  = var.pdAuthSvcToken #toke service pager duty
  }
}

#aqui define o scalation policy 
resource "newrelic_notification_channel" "default-pagerduty" {
  account_id     = var.accountId
  name           = "pagerduty-account-example"
  type           = "PAGERDUTY_SERVICE_INTEGRATION"
  destination_id = newrelic_notification_destination.default-pagerduty.id
  product        = "IINT"

  property {
    key   = "summary"
    value = "Created by IaC"
  }
}

#aqui define o canal de destino 
resource "newrelic_notification_channel" "nr-fin-sre-tmp-alerts" {
  account_id     = var.accountId
  name           = "fin-sre-tmp-alerts"
  type           = "SLACK"
  destination_id = var.slackDestinationId
  product        = "IINT"

  property {
    key   = "channelId"
    value = "C044P4PHURZ"
  }
}

resource "newrelic_workflow" "tmp-nr-sre" {
  name                  = "Financial SRE Workflow TRS" # mudou 
  account_id            = var.accountId
  enabled               = true
  enrichments_enabled   = true
  muting_rules_handling = "NOTIFY_ALL_ISSUES"
  issues_filter {
    name = "filter-name"
    type = "FILTER"
    predicate {
      attribute = "labels.policyIds"
      operator  = "EXACTLY_MATCHES"
      values    = [newrelic_alert_policy.policy1.id, newrelic_alert_policy.policy2.id]
    }
  }
  destination {
    channel_id = newrelic_notification_channel.nr-fin-sre-tmp-alerts.id
  }

  destination {
    channel_id = newrelic_notification_channel.default-pagerduty.id
  }
}
```

to reproduce
```
variables.tf
variable "accountId" {
}

variable "apiKey" {
}

variable "pdAuthSvcToken" {
}

variable "slackDestinationId" {
}
export vars:
export TF_VAR_accountId="" #new relic account id
export TF_VAR_apiKey="" #new relic api key
export TF_VAR_pdAuthSvcToken="" #pager duty service token
export TF_VAR_slackDestinationId="" #slack destination id
```

**Comment by juliostone** [comment]
up help

**Comment by IliaShurygin** [comment]
Hi! Thank you for the detailed report!

To sum it up, do I get it right that you are facing three main issues?
1. Issue filters are impossible to update though TF
2. Implicit channel deletion brakes TF state making it impossible to update anything anymore
3. Errors are not informative (you did mention this as a problem explicitly, but still..)

We are already working on (1), while (2) and (3) are on the radar! I cannot promise you a specific ETA, but the fixes should be rolled out in the foreseeable future, long before EOL date of the old solution (~Aug 2023).

**Comment by leonardoHaubrich** [comment]
@IliaShurygin hks for the reply, considering the EOL date, does it mean that even after the migration described in the image below, we will be able to create resources like "notification settings" without any problems?

This issue worries me because today we have all our policies and notifications settings as code, so after migration I'm afraid that the automations may break thanks to the resource change

![image](https://user-images.githubusercontent.com/54046696/195909232-f2cd38d5-cd0f-4fb6-9cf9-e5755ec2bae0.png)

**Comment by IliaShurygin** [comment]
@leonardoHaubrich, yes, the old resources are going to continue working till EOL.

In the UI side, you'll have to use the new workflows, but all changes to old resources done through TF are automatically and synchronously synced with the new resources.

**Comment by IliaShurygin** [comment]
The fix for the channel deletion has been released in v3.5.1

The other reported problem (filter updates) still exists. 
In order to reduce the number of duplicate issues, I am closing this ticket and I ask everyone to track resolution of the filter change problem via https://github.com/newrelic/terraform-provider-newrelic/issues/1986
